usecases/backup/backend.go

```diff
@@ -65,52 +65,55 @@ const (
 
 var _NUMCPU = runtime.NumCPU()
 
-type objStore struct {
-	b        modulecapabilities.BackupBackend
-	BasePath string
+type objectStore struct {
+	backend modulecapabilities.BackupBackend
+
+	backupId string // use supplied backup id
+	bucket   string // Override bucket for one call
+	path     string // Override path for one call
 }
 
-func (s *objStore) HomeDir() string {
-	return s.b.HomeDir(s.BasePath)
+func (s *objectStore) HomeDir(overrideBucket, overridePath string) string {
+	return s.backend.HomeDir(s.backupId, overrideBucket, overridePath)
 }
 
-func (s *objStore) WriteToFile(ctx context.Context, key, destPath string) error {
-	return s.b.WriteToFile(ctx, s.BasePath, key, destPath)
+func (s *objectStore) WriteToFile(ctx context.Context, key, destPath, overrideBucket, overridePath string) error {
+	return s.backend.WriteToFile(ctx, s.backupId, key, destPath, overrideBucket, overridePath)
 }
 
 // SourceDataPath is data path of all source files
-func (s *objStore) SourceDataPath() string {
-	return s.b.SourceDataPath()
+func (s *objectStore) SourceDataPath() string {
+	return s.backend.SourceDataPath()
 }
 
-func (s *objStore) Write(ctx context.Context, key string, r io.ReadCloser) (int64, error) {
-	return s.b.Write(ctx, s.BasePath, key, r)
+func (s *objectStore) Write(ctx context.Context, key, overrideBucket, overridePath string, r io.ReadCloser) (int64, error) {
+	return s.backend.Write(ctx, s.backupId, key, overrideBucket, overridePath, r)
 }
 
-func (s *objStore) Read(ctx context.Context, key string, w io.WriteCloser) (int64, error) {
-	return s.b.Read(ctx, s.BasePath, key, w)
+func (s *objectStore) Read(ctx context.Context, key, overrideBucket, overridePath string, w io.WriteCloser) (int64, error) {
+	return s.backend.Read(ctx, s.backupId, key, overrideBucket, overridePath, w)
 }
 
-func (s *objStore) Initialize(ctx context.Context) error {
-	return s.b.Initialize(ctx, s.BasePath)
+func (s *objectStore) Initialize(ctx context.Context, overrideBucket, overridePath string) error {
+	return s.backend.Initialize(ctx, s.backupId, overrideBucket, overridePath)
 }
 
 // meta marshals and uploads metadata
-func (s *objStore) putMeta(ctx context.Context, key string, desc interface{}) error {
+func (s *objectStore) putMeta(ctx context.Context, key, overrideBucket, overridePath string, desc interface{}) error {
 	bytes, err := json.Marshal(desc)
 	if err != nil {
-		return fmt.Errorf("marshal meta file %q: %w", key, err)
+		return fmt.Errorf("(putMeta)marshal meta file %q: %w", key, err)
```

Comment:
suggestion: it would be good to add space between `(putMeta)` and `marshal`, or could we change it to `puthMeta: ...`?